syntax = "proto3";
package bazel_template.proto;
import "src/proto/error.proto";
import "src/proto/enums.proto";

service MEGAService {
  // Load folder structure from server
  rpc LoadFolder(LoadFolderRequest) returns (LoadFolderResponse) {}
}

// Request message for LoadFolder
message LoadFolderRequest {
  OpCode op = 1;
  string folder_id = 2;  // The ID of the folder to load
}

// Response message for LoadFolder
message LoadFolderResponse {
  ErrCode err_code = 1;
  string folder_structure = 2;  // JSON string containing folder structure
  int32 status = 3;             // Status of the response (1 means last chunk)
}
